.github/workflows/python.yml

```diff
@@ -0,0 +1,47 @@
+name: Python
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  test:
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
+        python-version: ['3.12']
+    runs-on: ${{ matrix.os }}
+    env:
+      working-directory: ./bindings/python
+
+    defaults:
+      run:
+        working-directory: ${{ env.working-directory }}
+
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v3
+
+    - name: install rust stable
+      uses: dtolnay/rust-toolchain@stable
+
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v5
+      with:
+        python-version: ${{ matrix.python-version }}
+
+    - name: Install dependencies
+      run: pip install -r requirements.txt && pip install -r requirements-dev.txt
+
+    - name: Run Ruff check
+      run: ruff check limbo tests
```

Comment:
This job should probably run `ruff format --check`. However, Ruff configs seem to be incompatible with `format` and `check`. It would need checking the configs first.